### PR TITLE
show trainee schedule in reports

### PR DESCRIPTION
**Pull request by namframgia**

### CODE REVIEW

**Comment by namframgia** [comment]
https://edu-redmine.framgia.vn/issues/3463

File: app/Http/Resources/WeekScheduleResource.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * @param  \Illuminate\Http\Request  $request
+     * @return array
+     */
+    public function toArray($request)
+    {
+        $date = Carbon::now();
+        $date->setDate($this->year, $this->month, $this->day);
+
+        $status = null;
+
```
Comment (framgiaciv3) [inline app/Http/Resources/WeekScheduleResource.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.ControlStructures.SwitchDeclaration.WrongOpenercase</br>**Severity**: error</br>**File**: app/Http/Resources/WeekScheduleResource.php</br>**Line**: 24</br>CASE statements must be defined using a colon

File: app/Http/Resources/WeekScheduleResource.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        switch ($this->status) {
+            case 1: {
+                $status = config('api.status.short.' . $this->status);
+                break;
```
Comment (framgiaciv3) [inline app/Http/Resources/WeekScheduleResource.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.ControlStructures.SwitchDeclaration.WrongOpenercase</br>**Severity**: error</br>**File**: app/Http/Resources/WeekScheduleResource.php</br>**Line**: 28</br>CASE statements must be defined using a colon

File: app/Http/Resources/WeekScheduleResource.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            }
+            case 2: {
+                $status = config('api.status.short.' . $this->status);
+                break;
```
Comment (framgiaciv3) [inline app/Http/Resources/WeekScheduleResource.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.ControlStructures.SwitchDeclaration.WrongOpenercase</br>**Severity**: error</br>**File**: app/Http/Resources/WeekScheduleResource.php</br>**Line**: 32</br>CASE statements must be defined using a colon

File: app/Http/Resources/WeekScheduleResource.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+            }
+            case 3: {
+                $status = config('api.status.short.' . $this->status);
+                break;
```
Comment (framgiaciv3) [inline app/Http/Resources/WeekScheduleResource.php]: **Reporter**: CHECKSTYLE</br>**Rule**: PSR2.ControlStructures.SwitchDeclaration.WrongOpenerdefault</br>**Severity**: error</br>**File**: app/Http/Resources/WeekScheduleResource.php</br>**Line**: 36</br>DEFAULT statements must be defined using a colon

File: app/Repositories/ScheduleRepository.php [hunk 5]
```diff
@@ -60,7 +60,18 @@ public function showScheduleByWeek($id)
                 'day_month_id',
                 'dayMonth.id'
             )
-            ->where()
+            ->whereBetween('day', [
+                date('d', strtotime('-'.$day.' days')),
+                date('d', strtotime('+'.(6-$day).' days'))
```
Comment (framgiaciv3) [inline app/Repositories/ScheduleRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Repositories/ScheduleRepository.php</br>**Line**: 65</br>Comma required after last value in array declaration

File: app/Repositories/ScheduleRepository.php [hunk 6]
```diff
@@ -60,7 +60,18 @@ public function showScheduleByWeek($id)
...
+            ])
+            ->whereBetween('month', [
+                date('m', strtotime('-'.$day.' days')),
+                date('m', strtotime('+'.(6-$day).' days'))
```
Comment (framgiaciv3) [inline app/Repositories/ScheduleRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Repositories/ScheduleRepository.php</br>**Line**: 69</br>Comma required after last value in array declaration

File: app/Repositories/ScheduleRepository.php [hunk 7]
```diff
@@ -60,7 +60,18 @@ public function showScheduleByWeek($id)
...
+            ])
+            ->whereBetween('year', [
+                date('Y', strtotime('-'.$day.' days')),
+                date('Y', strtotime('+'.(6-$day).' days'))
```
Comment (framgiaciv3) [inline app/Repositories/ScheduleRepository.php]: **Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Arrays.ArrayDeclaration.NoCommaAfterLast</br>**Severity**: error</br>**File**: app/Repositories/ScheduleRepository.php</br>**Line**: 73</br>Comma required after last value in array declaration

File: app/Repositories/ScheduleRepository.php [hunk 8]
```diff
@@ -60,7 +60,18 @@ public function showScheduleByWeek($id)
...
+                date('d', strtotime('-'.$day.' days')),
+                date('d', strtotime('+'.(6-$day).' days')),
+            ])
+            ->whereBetween('month', [
+                date('m', strtotime('-'.$day.' days')),
+                date('m', strtotime('+'.(6-$day).' days')),
+            ])
+            ->whereBetween('year', [
+                date('Y', strtotime('-'.$day.' days')),
+                date('Y', strtotime('+'.(6-$day).' days')),
```
Comment (BunnyPi04) [inline app/Repositories/ScheduleRepository.php]: convention

File: app/Http/Resources/WeekScheduleResource.php [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+            case 2:
+                $status = config('api.status.short.' . $this->status);
+                break;
+            case 3:
+                $status = config('api.status.short.' . $this->status);
+                break;
+            default:
+                $status = config('api.status.short.' . $this->status);
```
Comment (BunnyPi04) [inline app/Http/Resources/WeekScheduleResource.php]: các case xử lý giống nhau thì switch case làm gì hả em (??)

**Pull request merged**